src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -499,11 +546,23 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await singleton.AcquireAsync(functionCancellationTokenSource.Token);
             }
 
-            var timer = StartFunctionTimeout(instance, globalFunctionTimeout, functionCancellationTokenSource, traceWriter);
+            bool hasTimerElapsed = false;
+            var timer = StartFunctionTimeout(instance, timeoutConfig.Timeout, timeoutConfig.TimeoutWhileDebugging, functionCancellationTokenSource, traceWriter, () => hasTimerElapsed = true);
             try
             {
                 await invoker.InvokeAsync(invokeParameters);
             }
+            catch (OperationCanceledException)
+            {
+                // if cancellationToken is not signaled, it means this is a timeout
+                if (timeoutConfig.ThrowExceptionOnFunctionTimeout && hasTimerElapsed)
+                {
+                    string errorMessage = string.Format("Timeout while executing function: {0}", instance.FunctionDescriptor.ShortName);
```

Comment:
The timeout value is already logged elsewhere, but I've added the Timeout to both the message and a TImeSpan value to the exception itself. It may be useful later, so might as well do it now.

---
In reply to: [76689057](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r76689057) [](ancestors = 76689057)